src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs

```diff
@@ -37,38 +36,52 @@ private static Version ReadCore(ref Utf8JsonReader reader)
             Debug.Assert(reader.TokenType is JsonTokenType.PropertyName or JsonTokenType.String);
 
 #if NET
-            if (!JsonHelpers.IsInRangeInclusive(reader.ValueLength, MinimumVersionLength, MaximumEscapedVersionLength))
+            int bufferLength = reader.ValueLength;
+            char[]? rentedBuffer = null;
+            try
             {
-                ThrowHelper.ThrowFormatException(DataType.Version);
+                Span<char> charBuffer = bufferLength <= JsonConstants.StackallocCharThreshold
+                    ? stackalloc char[JsonConstants.StackallocCharThreshold]
+                    : (rentedBuffer = ArrayPool<char>.Shared.Rent(bufferLength));
+
+                int bytesWritten = reader.CopyString(charBuffer);
+                ReadOnlySpan<char> source = charBuffer.Slice(0, bytesWritten);
+
+                if (source.Length > 0 && (char.IsWhiteSpace(source[0]) || char.IsWhiteSpace(source[^1]) || source[0] == '+'))
```

Comment:
Please remove the check for the '+' digit as we only want to enforce leading and trailing whitspace. Do this in the other branch as well.